test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1

```diff
@@ -5,219 +5,460 @@
+            $result =  Format-Hex $inputFile1
+
+            $result | Should Not BeNullOrEmpty
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $actualResult = $result.ToString()
+            ($actualResult -match $inputText1) | Should Be $true
         }
-    }
 
-    # This test ensures that if we stream bytes from a file, the output is correct
-    It "ValidateStreamOfBytesFromFileHasProperOutput" {
-        $result = Get-Content $InputFile1 -Encoding Byte | Format-Hex
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        if ( $IsCoreCLR ) {
-            $result.ToString() | Should be    "00000000   48 65 6C 6C 6F 20 57 6F 72 6C 64                 Hello World     "
+        It "Validate file input from Pipeline 'Get-ChildItem `$inputFile1 | Format-Hex'" {
+
+            $result = Get-ChildItem $inputFile1 | Format-Hex
+
+            $result | Should Not BeNullOrEmpty
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $actualResult = $result.ToString()
+            ($actualResult -match $inputText1) | Should Be $true
         }
-    }
 
-    # This test is to validate the alias for Format-Hex cmdlet.
-    It "ValidateCmdletAlias" {
+        It "Validate that streamed text does not have buffer underrun problems ''a' * 30 | Format-Hex'" {
+
+            $result = "a" * 30 | Format-Hex
 
-        try
-        {
-            $result = Get-Command fhx -ErrorAction Stop
             $result | Should Not BeNullOrEmpty
-            $result.CommandType.ToString() | Should Be "Alias"
+            ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
+            $actualResult = $result.ToString() -split "`r`n"
+            $actualResult.Count | should be 2
+            $actualResult[0].ToString() | Should be "00000000   61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61  aaaaaaaaaaaaaaaa"
+            $actualResult[1].ToString() | Should be "00000010   61 61 61 61 61 61 61 61 61 61 61 61 61 61        aaaaaaaaaaaaaa  "
         }
-        catch
-        {
-            $_ | Should BeNullOrEmpty
+
+        It "Validate that files do not have buffer underrun problems 'Format-Hex -path `$InputFile2'" {
+
+            $result = Format-Hex -path $InputFile2
+
+            $result | Should Not BeNullOrEmpty
+            $result.Count | should be 2
+            $result[0].ToString() | Should be "00000000   54 68 69 73 20 69 73 20 61 20 62 69 74 20 6D 6F  This is a bit mo"
+            if ( $IsCoreCLR ) {
+                $result[1].ToString() | Should be "00000010   72 65 20 74 65 78 74                             re text         "
+            }
+        }
+
+        It "Validate the alias for Format-Hex cmdlet 'Get-Command fhx -ErrorAction Stop'" {
+
+            try
```

Comment:
This test case could probably be simplified to:

{ Get-Alias fhx } | Should Not Throw